### PR TITLE
Add initial danger CI script.

### PR DESCRIPTION
**Pull request by jverkoey**
Adds support for running Danger checks against PRs.

## What is Danger?

[Danger](http://danger.systems/js/) is a tool built and maintained by @orta at Artsy. From the site docs:

> Danger runs during your CI process, and gives teams the chance to automate common code review chores.
> 
> This provides another logical step in your build, through this Danger can help lint your rote tasks in daily code review.
> 
> You can use Danger to codify your teams norms. Leaving humans to think about harder problems.
> 
> This happens by Danger leaving messages inside your PRs based on rules that you create with JavaScript or TypeScript.
> 
> Over time, as rules are adhered to, the message is amended to reflect the current state of the code review.

## How does it work?

Danger runs as part of our CI workflow. Danger has a dedicated job that runs in parallel with our other kokoro jobs. When the kokoro job runs, it invokes our `.kokoro` script with `danger` as the environment. The danger "logic" is found in Dangerfile.js. To start, our dangerfile simply checks whether a PR has a description that is longer than 10 characters. Over time, I hope that we can expand to more checks such as:

- Recommending that source changes be matched by test changes.
- Reminding to regenerate the component README.md if any of the doc articles change.
- Reminding that podspec and BUILD file changes often have to go hand-in-hand.
- Enforcing use of umbrella header imports over direct header imports.

## Can I run the danger checks locally?

Danger runs on PRs, so you won't be able to preview warnings before pushing your changes to GitHub. You technically can run the danger CI locally, but it will likely do the same work as the kokoro job (which starts running once you push the PR).

## What happens if danger gives a warning and I don't have any new changes to push?

For now, Danger is simply a warning, so it won't block any PRs from landing. But if you want to make sure that your PR is clean you can restart the job by clicking the Danger status bit in the PR and restarting the job for your PR.

### CODE REVIEW

File: .kokoro
Comment (romoore) [inline .kokoro:199]: Is "YEP" a documented value for this variable in Danger?

File: yarn.lock
Comment (romoore) [inline yarn.lock]: There are a lot of downstream dependencies here. Have they been reviewed for license approval?

**Review by romoore — DISMISSED** [review]
Mostly concerned about the licensing of the downstream dependencies.

File: .kokoro
Comment (jverkoey) [reply .kokoro:199]: http://danger.systems/js/guides/the_dangerfile.html

File: yarn.lock
Comment (jverkoey) [reply yarn.lock]: I've removed the yarn.lock file. This is only needed for CI.

File: .kokoro
Comment (jverkoey) [reply .kokoro:199]:
```
You can manually trigger danger against a pull request on the command line by setting the following environmental variables:

export DANGER_FAKE_CI="YEP"
export DANGER_TEST_REPO='username/reponame'
```

**Review by romoore — COMMENTED** [review]
Even if we remove the lock file, isn't it going to install all that software when we run it? Have we done a license review?

**Comment by jverkoey** [comment]
This isn't part of an Android app, an iOS app, a desktop app, or consumer hardware, so I don't believe this falls under the requirements for doing license reviews.

**Comment by yarneo** [comment]
Thinking of this more from the restriction side; i’ve seen very small PRs that the PR title was sufficient to explain the context and no description was needed. This will just make the developer have to copy paste the title into the description box. Have you been witnessing PRs often that have no description and therefore didn’t provide enough information to understand what the author was intending?

**Comment by jverkoey** [comment]
Can you link to an example of a PR that didn't need further context?

**Comment by jverkoey** [comment]
And note: Danger only posts a comment to the PR thread with the listed warnings. These warnings are not blocking and serve primarily as informational notices.

**Comment by yarneo** [comment]
https://github.com/material-components/material-components-ios/pull/3677

**Comment by jverkoey** [comment]
Cool - yeah in those cases I think this tool will look a bit noisier, but those are likely the exception rather than the norm.

**Comment by romoore** [comment]
@jverkoey My concern would be the *use* of the software. If one of the dependencies has an EULA or similar that puts restrictions on the project or maintainers, that's where I'd be concerned.

@yarneo I think even in the case of #3677, a quick description doesn't hurt. For example, why was it previously absent from the README? Is it being changed from a static image to a gif?  Was it a .mov and now it's a gif?

**Comment by orta** [comment]
Hey all, with respect to checking the dependencies of this, you can use a copy of [npm-license](https://github.com/AceMetrix/npm-license) in this repo to see what it adds, ( it should be mostly MIT, ISC and BSD, then maybe one or two [WTFPLs](https://en.wikipedia.org/wiki/WTFPL) (:-/) )

`<edit/>` perfect, no WTFPLs

**Comment by jverkoey** [comment]
Licenses in use by danger's dependencies:

```
License: Apache-2.0
License: BSD
License: ISC
License: MIT
```

**Comment by jverkoey** [comment]
These licenses are all pretty standard for us and don't include any usage restrictions.

**Comment by jverkoey** [comment]
I've changed the script to check for a `[ ]` notation in our PR instead, as this is already an accepted cultural trait of our PRs.

File: dangerfile.js
Comment (romoore) [inline dangerfile.js]: What do you we do for multiple components? Previously, I might omit the component in the summary and just tag the components with labels.

```
[ActivityIndicator, BottomNavigation, Buttons, ButtonBar,  Slider, Tabs, Text Fields] Update BUILD file `includes` paths
```

**Review by yarneo — APPROVED** [review]

File: dangerfile.js
Comment (jverkoey) [reply dangerfile.js]: If something is affecting multiple components and the change can't be broken up into multiple PRs, then the common trait of the change could be used in the PR title. E.g. `[documentation]`.

Comment (romoore) [reply dangerfile.js]: Ok. I wasn't sure how to interpret 'component' below - I thought it had to map to a specific component.

Comment (jverkoey) [reply dangerfile.js]: Ah that's fair. I'll flesh out the description in a follow-up PR.

Comment (jverkoey) [reply dangerfile.js]: https://github.com/material-components/material-components-ios/pull/3757

**Pull request merged**
